journal.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import os
+import datetime
+from pyramid.config import Configurator
+from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from pyramid.security import remember, forget
+from waitress import serve
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import scoped_session, sessionmaker
+from zope.sqlalchemy import ZopeTransactionExtension
+from sqlalchemy.exc import DBAPIError
+from cryptacular.bcrypt import BCRYPTPasswordManager
+
+
+DBSession = scoped_session(sessionmaker(
+            extension=ZopeTransactionExtension()))
+
+Base = declarative_base()
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql://tyler:tyler@localhost:5432/learning-journal'
+)
+
+
+class Entry(Base):
+    __tablename__ = 'entries'
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(127), nullable=False)
+    text = sa.Column(sa.UnicodeText, nullable=False)
+    created = sa.Column(
+        sa.DateTime, nullable=False,
+        default=datetime.datetime.utcnow
+    )
+
+    @classmethod
+    def write(cls, title=None, text=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, text=text)
+        session.add(instance)
+        return instance
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.created.desc()).all()
+
+
+def do_login(request):
+    username = request.params.get('username', None)
+    password = request.params.get('password', None)
+    if not (username and password):
+        raise ValueError('both username and password are required')
+    settings = request.registry.settings
+    manager = BCRYPTPasswordManager()
+    if username == settings.get('auth.username', ''):
+        hashed = settings.get('auth.password', '')
+        return manager.check(hashed, password)
+    return False
+
+
+@view_config(route_name='other', renderer='string')
+def other(request):
+    #    import pdb; pdb.set_trace()
+    return request.matchdict
+
+
+# @view_config(route_name='home', renderer='templates/list.jinja2')
+# def list_view(request):
+#     entries = Entry.all()
+#     return {'entries': entries}
+@view_config(route_name='home', renderer='templates/base.jinja2')
+def home(request):
+    entries = Entry.all()
+    return {'entries': entries}
+
+
+@view_config(route_name='add', renderer='templates/add.jinja2')
+def add(request):
+    entries = Entry.all()
+    return {'entries': entries}
```

Comment:
Do you need to hand a list of entries to a page that shows an 'add' form?  Why are these there?  What do they do?